src/writers/json-writer.cc

```diff
@@ -0,0 +1,101 @@
+namespace std { class type_info; }
+
+#include <reporter.hh>
+#include <file-parser.hh>
+#include <configuration.hh>
+#include <writer.hh>
+#include <utils.hh>
+
+#include <string>
+#include <list>
+#include <unordered_map>
+#include <iostream>
+#include <fstream>
+
+#include "writer-base.hh"
+
+using namespace kcov;
+
+class JsonWriter : public WriterBase
+{
+public:
+	JsonWriter(IFileParser &parser, IReporter &reporter,
+			const std::string &outFile) :
+		WriterBase(parser, reporter),
+		m_outFile(outFile)
+	{
+	}
+
+	void onStartup()
+	{
+	}
+
+	void onStop()
+	{
+	}
+
+	void write()
+	{
+		std::ofstream out(m_outFile);
+
+		// Output directory not writable?
+		if (!out.is_open())
+			return;
+
+		IConfiguration &conf = IConfiguration::getInstance();
+		unsigned int nExecutedLines = 0;
+		unsigned int nCodeLines = 0;
+
+		setupCommonPaths();
+
+		for (FileMap_t::const_iterator it = m_files.begin();
+				it != m_files.end();
+				++it) {
+			File *file = it->second;
+			nCodeLines += file->m_codeLines;
+			nExecutedLines += file->m_executedLines;
```

Comment:
So, when I looked at the sonarqube and cobertura writers, the lines were counted by iterating over every line in the file and incrementing a counter.

I copied this out of the HTML writer since it doesn't _appear_ to loop over line, but merely accesses these same properties and sums them from each file (which is what i'd like to do). 

I can definitely try to just iterate over the lines in each file anyway, since I want to do that for the file by file synopsis I plan to add this report; it just felt a bit weird that this didn't work and I couldn't immediately see why.